nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -200,7 +200,7 @@ fi
 # Ask the user to set a root password, but only if the passwd command
 # exists (i.e. when mutable user accounts are enabled).
 if [[ -z $noRootPasswd ]] && [ -t 0 ]; then
-    if nixos-enter --root "$mountPoint" -c 'test -e /nix/var/nix/profiles/system/sw/bin/passwd'; then
+    if jq -e '.mutableUsers' "$system/users-groups.json" > /dev/null; then
```

Comment:
this should also have a fallback to cover when the json file doesnt exist, for users installing an old nixpkgs with a new nixos-install

Reply:
Great point, thanks Clever. I made the logic here a bit more robust against that.